strawberryfields/tdm/tdmprogram.py

```diff
@@ -392,12 +381,18 @@ def __init__(self, N, name=None):
         super().__init__(num_subsystems=self.concurr_modes, name=name)
 
         self.type = "tdm"
+        self._is_space_unrolled = False
+
         self.timebins = 0
         self.spatial_modes = 0
         self.measured_modes = []
         self.rolled_circuit = None
         # `unrolled_circuit` only contains the unrolled single-shot circuit
         self.unrolled_circuit = None
+        # `space_unrolled_circuit` only contains the space-unrolled single-shot circuit
```

Comment:
It would be great to get an overall description of the idea of space unrolling and to have a comparison with unrolling. This could maybe be done as an example and the change in the program could be explained.